library/Fission/URL/Subdomain.hs

```diff
@@ -0,0 +1,21 @@
+-- | URL helpers
+module Fission.URL.Subdomain
+  ( normalizePrefix
+  , prefix
+  , module Fission.URL.Subdomain.Types
+  ) where
+
+import Fission.Prelude
+import Fission.URL.Subdomain.Types
+import Fission.URL.DomainName.Types
+
+-- | Prefix a domain named with an optional subdomain
+normalizePrefix :: DomainName -> Maybe Subdomain -> DomainName
+normalizePrefix domain = \case
```

Comment:
Very minor, but: 
when a whole function is wrapped in a `case` statement, my brain parses it better with something like
```
normalizePrefix domain Nothing = domain
normalizePrefix domain (Just subdomain) = prefix domain subdomain
```
Thoughts? Any reason to prefer one over the other?

Reply:
Yeah, they're the _exact same_ to the compiler, so this is totally a style thing. There's no community consensus on this. I have a few reasons for preferring the `\case` version.

1. It saves a few keystrokes. Whatever.
2. It's consistent
  * `Maybe` has two branches, but we have places where it's much more
    * For example, the different kinds of IPFS errors
  * We pipe into `\case` a fair bit inside functions
3. It's clear that we're switching on the last argument as logic
4. It's still clear from the type signature how many arguments there are

This is purely style preference. I also realize that I can read point-free code pretty natively, which is not typical of most devs. Maybe we can make a style choice like "if there's 1-3 branches, then use function heads. More than 3 and use `\case`".

Any preference / thoughts?

Reply:
yeah no huge preference here. I'm justed to seeing `\case` within functions so I think it just confused me a bit to see the last argument being cased on like that. 

Didn't know the term "point-free code". Yup that definitely sums up why I had a hard initial time parsing this

I still have a slight preference for using function heads just because I find them more immediately intuitive, but it's slight so really no need to adapt to that.

Curious tho, why would `\case` be better with more branches? Just because you don't have to keep writing the function name & other params?

Reply:
> Curious tho, why would \case be better with more branches? Just because you don't have to keep writing the function name & other params?

Yes, exactly. Like, this would be annoying to type out:

```haskell
checkIn name colour tasks Monday    = -- ...
checkIn name colour tasks Tuesday   = -- ...
checkIn name colour tasks Wednesday = -- ...
checkIn name colour tasks Thursday  = -- ...
checkIn name colour tasks Friday    = -- ...
checkIn name colour tasks Saturday  = -- ...
checkIn name colour tasks Sunday    = -- ...

-- versus

checkIn name colour tasks = \case
  Monday    -> -- ...
  Tuesday   -> -- ...
  Wednesday -> -- ...
  Thursday  -> -- ...
  Friday    -> -- ...
  Saturday  -> -- ...
  Sunday    -> -- ...
```